lib/bundle/utils.rb

```diff
@@ -24,6 +24,18 @@ def self.cask_installed?
     @@cask ||= Kernel.system("brew cask --version &>/dev/null")
   end
 
+  def self.installed_formulae
+    @@installed_formulae ||= `brew list -1`.split("\n")
+  end
+
+  def self.outdated_formulae
+    @@outdated_formulae ||= `brew outdated --quiet`.split("\n")
+  end
+
+  def self.pinned_formulae
+    @@pinned_formulae ||= `brew list --pinned`.split("\n")
+  end
```

Comment:
@xu-cheng btw, as I understand external commands like `brew bundle` has full access to the Homebrew API https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/External-Commands.md#ruby-commands

Could we try to use Homebrew Ruby API instead of system calls?